battleship.py

```diff
@@ -14,38 +14,35 @@ def retrieve_size():
 
 def clear_grid():
     new_grid = []
-    for i in range(16):
+    for i in range(grid_size):
```

Comment:
I assume you are grabbing grid_size from a global variable and while that works it's much better practice to make it a parameter in the function and pass it the value when you call it. 

demonstration
```
def clear_grid(grid_size):
...

grid_size = 20

clear_grid(grid_size)
```